### DISCUSSION TITLE
how to fix this problem with physx on FreeBSD?

### DISCUSSION THREAD

**Discussion by Martinfx** [RavEngine/RavEngine#24] [Q&A]
Why do you use physx and not https://github.com/jrouwe/JoltPhysics

```
PhysX Build Platform: 
Using CXX Compiler: /usr/bin/c++
PHYSX ROOT /home/maxfx/Documents/RavEngine/deps/physx/physx
PhysX Build Platform: 
Using CXX Compiler: /usr/bin/c++
CMake Error at deps/physx/externals/cmakemodules/GetCompilerAndPlatform.cmake:14 (SET):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  deps/physx/externals/cmakemodules/GetCompilerAndPlatform.cmake:98 (GetCompiler)
  deps/physx/externals/cmakemodules/NvidiaBuildOptions.cmake:99 (GetPlatformBinName)
  deps/physx/physx/source/compiler/cmake/CMakeLists.txt:69 (INCLUDE)

```

**Comment by Ravbug** [comment]
Hey, thanks for your interest! This project does not support FreeBSD. That said, you could probably add support for it in your own fork.  

To fix this specific problem, you most likely need to enable these CMake statements on both Linux and FreeBSD. Currently they're only enabled for Linux. 
https://github.com/RavEngine/RavEngine/blob/19c13c3ae97ebc5422f24b8e9023867916c9d73d/CMakeLists.txt#L361-L364

There are probably more changes that need to be made to get FreeBSD to work, since I put exactly zero thought into making FreeBSD work when writing the code for this project, and I suspect the story is similar for many of the third party libraries this project uses. You're welcome to PR your changes if you get it working, but I can't promise that it won't break again in the future.

**Reply by Martinfx** [reply]
The problem is Physx 5 doesn't support FreeBSD. Physx 4 I am ported to FreeBSD. Are you going to convert from physx to joltphysics ?

**Reply by Ravbug** [reply]
> Are you going to convert from physx to joltphysics ?

no. But you are welcome to replace PhysX with Jolt in your own fork. I probably won't accept a PR for that though, sorry.
